docs/network-restrictions.md

```diff
@@ -0,0 +1,34 @@
+# Running behind a proxy
+
+If nodes only have access via a HTTP proxy, you can let the machine-controller configure all new nodes to use this proxy.
+For this the following flag must be set on the machine-controller side:
+```bash
+-node-http-proxy="http://192.168.1.1:3128"
+```
+This will set the following environment variables via /etc/environment on all nodes (lower & uppercase):
+- HTTP_PROXY
+- HTTPS_PROXY
+- NO_PROXY (Will always be set to `localhost,127.0.0.1`)
```

Comment:
I think being able to add hosts here will be required in most envs. We could probably also have a smarter default, e.G: all rfc1918 ranges and the link-local ranges. WDYT @thz?

Reply:
Indeed, we already have examples of using NO_PROXY

Reply:
@kron4eg Do you have a link to such an example ?

Reply:
@kron4eg Maybe just post the potential content of that field instead of a link ?

Reply:
Having all rfc1918 ranges in NO_PROXY by default is perhaps a bit too much. But 127.0.0.1 and link-local makes sense. Perhaps also the k8s cluster domain(s).

Reply:
Done with https://github.com/kubermatic/machine-controller/pull/559/commits/649786bd2d98d7c54b81975417e047a9a654ab42

Reply:
@thz Do you have an example usecase where it does make sense to not have the rfc1918 networks in no_proxy?

Reply:
of course. let's say you are in a 10.1.0.0/16 CIDR vpc and you want to talk to that other 10.2.0.0/16 vpc directly by aws peering.

Reply:
Aehm. Not _directly_ but via the proxy.